celery/app/task.py

```diff
@@ -769,11 +774,14 @@ def apply(self, args=None, kwargs=None,
             task.name, task, eager=True,
             propagate=throw, app=self._get_app(),
         )
+
         ret = tracer(task_id, args, kwargs, request)
         retval = ret.retval
+        state = states.SUCCESS if ret.info is None else ret.info.state
         if isinstance(retval, ExceptionInfo):
             retval, tb = retval.exception, retval.traceback
-        state = states.SUCCESS if ret.info is None else ret.info.state
```

Comment:
Not sure this line change is necessary, I can't seem to find any effect. Would you mind reverting this, if not needed?